File: Prueba Tecnica Crazy Image Software/demo/src/main/java/crazyImage/demo/model/Task.java
package crazyImage.demo.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

@Entity
@Table(name = "task")
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long Id;

    @NotNull(message = "title cannot be null")
    @NotBlank(message = "title cannot be blank")
    private String title;

    @NotNull(message = "description cannot be null")
    @NotBlank(message = "description cannot be blank")
    private String description;

    @NotNull(message = "status cannot be null")
    @NotBlank(message = "status cannot be blank")
    private String status = "PENDING";

    @NotNull(message = "user_Id cannot be null")
    private Long user_Id;

    public Task(Long id, String title, String description, String status, Long user_Id) {
        Id = id;
        this.title = title;
        this.description = description;
        this.status = status;
        this.user_Id = user_Id;
    }

    public Task() {
    }

    public Long getId() {
        return Id;
    }

    public void setId(Long id) {
        Id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Long getUser_Id() {
        return user_Id;
    }

    public void setUser_Id(Long user_Id) {
        this.user_Id = user_Id;
    }

}

File: Prueba Tecnica Crazy Image Software/FrontendCrazyImage/src/types/task.ts
export interface Task {
    id: number;
    title: string;
    description: string;
    status: 'PENDING' | 'IN_PROGRESS' | 'COMPLETED';
    user_Id: number;
    user?: User;
}

export interface User {
    id?: number;
    email?: string;
    username?: string;
}

export interface CreateTaskRequest {
    title: string;
    description: string;
    status: 'PENDING' | 'IN_PROGRESS' | 'COMPLETED';
    user_Id: number;
}

export interface UpdateTaskRequest {
    title?: string;
    description?: string;
    status?: 'PENDING' | 'IN_PROGRESS' | 'COMPLETED';
    user_Id?: number;
}